community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/TransactionStateMachine.java

```diff
@@ -112,10 +118,7 @@ public void reset() throws TransactionFailureException
 
     private void after()
     {
-        if ( ctx.currentTransaction != null )
```

Comment:
It is safe to remove the null check because it wasn't needed to begin with - is that correct?

Reply:
Besides its redundancy, when an error occurs in `streamResult` call - the call may begin with a valid `ctx.currentTransaction` but end with a `null` one which will leave that specific transaction left as referenced by ThreadLocal variables.

I removed the null check as a `remove the ThreadLocal reference at all times` kind of behavior.